### PR TITLE
[bitnami/elasticsearch] attach PVC's for es master node data

### PR DESCRIPTION
**Pull request by sameersbn**
**Description of the change**
Elasticsearch master nodes store the cluster status at `/bitnami/elasticsearch/data`. Among other things this includes the UUID of the elasticsearch cluster. Without a persistent data store for this data, the UUID of a cluster could change if k8s node(s) hosting the es master nodes go down and are scheduled on some other master node. In the event that this happens, the data nodes will no longer be able to join a cluster as the uuid changed resulting in a broken cluster.

To resolve such issues, this PR attaches PVC's to the master nodes data store.

Additionally, the PR also adds helpers for setting the apiVersion for deployments and statefulsets based on the k8s version in use

**Benefits**

Avoids a very serious issue with the usability of the chart.

**Possible drawbacks**

None

**Applicable issues**

Encountered in internal testing

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

File: bitnami/elasticsearch/README.md [hunk 1]
```diff
@@ -455,6 +460,18 @@ You can enable this initContainer by setting `volumePermissions.enabled` to `tru
...
+
+Elasticsearch master nodes store the cluster status at `/bitnami/elasticsearch/data`. Among other things this includes the UUID of the elasticsearch cluster. Without a persistent data store for this data, the UUID of a cluster could change if k8s node(s) hosting the es master nodes go down and are scheduled on some other master node. In the event that this happens, the data nodes will no longer be able to join a cluster as the uuid changed resulting in a broken cluster.
+
+To resolve such issues, PVC's are now attached for master node data persistence.
+
+---
+
+Helm performs a lookup for the object based on its group (apps), version (v1), and kind (Deployment). Also known as its GroupVersionKind, or GVK. Changing the GVK is considered a compatibility breaker from Kubernetes' point of view, so you cannot "upgrade" those objects to the new GVK in-place. Earlier versions of Helm 3 did not perform the lookup correctly which has since been fixed to match the spec.
+
+In 4dfac075aacf74405e31ae5b27df4369e84eb0b0 the `apiVersion` of the deployment resources was updated to `apps/v1` in tune with the api's deprecated, resulting in compatibility breakage.
```
Comment (juan131) [inline bitnami/elasticsearch/README.md]: I would include a link to the commit:
```diff
- In 4dfac075aacf74405e31ae5b27df4369e84eb0b0 the ...
+ In [4dfac075aacf74405e31ae5b27df4369e84eb0b0](https://github.com/bitnami/charts/commit/4dfac075aacf74405e31ae5b27df4369e84eb0b0) the..
```

File: bitnami/elasticsearch/templates/master-statefulset.yaml [hunk 2]
```diff
@@ -32,9 +32,10 @@ spec:
       securityContext:
         fsGroup: {{ .Values.master.securityContext.fsGroup }}
       {{- end }}
-      {{- if .Values.sysctlImage.enabled }}
-      ## Image that performs the sysctl operation to modify Kernel settings (needed sometimes to avoid boot errors)
+      {{- if or .Values.sysctlImage.enabled (and .Values.volumePermissions.enabled .Values.master.persistence.enabled) }}
       initContainers:
+      {{- if .Values.sysctlImage.enabled }}
```
Comment (juan131) [inline bitnami/elasticsearch/templates/master-statefulset.yaml]: Indent this `if` a couple of spaces

File: bitnami/elasticsearch/templates/master-statefulset.yaml [hunk 3]
```diff
@@ -46,6 +47,25 @@ spec:
...
+            - |
+              chown -R {{ .Values.securityContext.runAsUser }}:{{ .Values.securityContext.fsGroup }} //bitnami/elasticsearch/data
+          securityContext:
+            runAsUser: 0
+          {{- if .Values.volumePermissions.resource }}
+          resources: {{- toYaml .Values.volumePermissions.resources | nindent 12 }}
+          {{- end }}
+          volumeMounts:
+            - name: data
+              mountPath: "/bitnami/elasticsearch/data"
```
Comment (juan131) [inline bitnami/elasticsearch/templates/master-statefulset.yaml]: I don't think you need to doublequote the mountPath

File: bitnami/elasticsearch/values.yaml [hunk 4]
```diff
@@ -427,23 +462,26 @@ ingest:
...
-  #  periodSeconds: 10
-  #  timeoutSeconds: 5
-  #  successThreshold: 1
-  #  failureThreshold: 5
+    enabled: true
+    initialDelaySeconds: 90
+    periodSeconds: 10
+    timeoutSeconds: 5
+    successThreshold: 1
+    failureThreshold: 5
```
Comment (juan131) [inline bitnami/elasticsearch/values.yaml]: Why did you uncomment this block (and others probes)?

Comment (sameersbn) [reply bitnami/elasticsearch/values.yaml]: the probes use these as the defaults https://github.com/bitnami/charts/blob/master/bitnami/elasticsearch/templates/master-statefulset.yaml#L84-L102

**Pull request merged**